### PR TITLE
19025 Sortable and accessible ui select

### PR DESCRIPTION
**Pull request by 0xF013**
[`* `#19025 ` NOT POSSIBLE TO CHANGE ORDER OF COLUMNS VIA "COLUMNS" MODAL`](http://community.openproject.org/work_packages/19025)

I enabled the sorting, please validate if it's not too much pixelhunting. I can work on that. 

This PR also includes a switch to the new UI select enhanced with my old accessibility stuff.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://0xf013-sortable-and-accessible-ui-select.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://0xf013-sortable-and-accessible-ui-select.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by ulferts** [comment]
@0xF013 I found it quite hard to actually change the order as there is only a very tiny space where I can drop the element. In the old select2, when using drag&drop, the elements would push the other elements out of the way and by that allow for more space where I can drop the element.
Additionally, I was never once able to move an element to the end of the list.

**Comment by 0xF013** [comment]
@ulferts so ui.select has this sorting where you is counts if the cursor with the dragged choice is on the left half or the right half of another choice and based on that it decides to drop it either in front or after the receiving choice

**Comment by ulferts** [comment]
@0xF013 sorting is looking good now except that I am still not able to sort anything to the back of the list making it the last element.

**Comment by 0xF013** [comment]
@ulferts did you try dropping a choice over the right half of the last choice?

**Comment by ulferts** [comment]
@0xF013 ah, that works. But then the "+" icons should only be displayed when I am over a functioning drop zone. Currently it is displayed for the whole back of the list which is not a valid drop zone.

**Comment by 0xF013** [comment]
@ulferts I'll try to fix some css around that

**Comment by ulferts** [comment]
@0xF013 any update on this one?

**Comment by 0xF013** [comment]
@ulferts will fix this today along with Jan's branch

**Comment by ulferts** [comment]
Thanks for the info @0xF013

**Comment by floriank** [comment]
@0xF013 please rebase.

**Comment by ulferts** [comment]
@0xF013 given our other todos it is good enough to merge. I personally would have liked to have some space included in the drop zone but that will have to be done later. Can you please rebase/merge so that I can press the button?

**Comment by 0xF013** [comment]
@ulferts jawohl

**Comment by ulferts** [comment]
Seems like I definitely need this T-Shirt: http://www.katzundgoldt.de/ru_befehlscharakter.htm

File: app/assets/stylesheets/content/ui_select.sass [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// keep this rule if you want ui-select drag-n-drop to work properly
```
Comment (myabc) [inline app/assets/stylesheets/content/ui_select.sass:1]: @0xF013 partials should begin with an underscore.

**Comment by 0xF013** [comment]
@myabc ack

**Comment by myabc** [comment]
@0xF013 quack

**Comment by 0xF013** [comment]
@myabc Quark

**Pull request merged**